Tests/Benchmarks/AnnulusBoundsBenchmark.cpp

```diff
@@ -0,0 +1,185 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2017-2018 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <algorithm>
+#include <chrono>
+#include <fstream>
+#include <functional>
+#include <iostream>
+#include <random>
+#include <vector>
+
+#include "Acts/Surfaces/AnnulusBounds.hpp"
+#include "Acts/Tests/CommonHelpers/BenchmarkTools.hpp"
+#include "Acts/Utilities/Definitions.hpp"
+#include "Acts/Utilities/Helpers.hpp"
+#include "Acts/Utilities/Units.hpp"
+
+using namespace Acts;
+
+int main(int /*argc*/, char** /*argv[]*/) {
+  std::mt19937 rng(42);
+  std::uniform_real_distribution<double> xDist(-2, 20);
+  std::uniform_real_distribution<double> yDist(-2, 20);
+
+  std::ofstream os{"annulus.csv"};
+  os << "x,y,inside_abs,inside_tol0,inside_tol1,inside_tol01,inside_cov"
+     << std::endl;
+
+  // === PROBLEM DATA ===
+
+  // bounds definition
+  double minRadius = 7.2;
+  double maxRadius = 12.0;
+  double minPhi = 0.74195;
+  double maxPhi = 1.33970;
+  Vector2D offset(-2., 2.);
+  AnnulusBounds aBounds(minRadius, maxRadius, minPhi, maxPhi, offset);
+
+  // helper to convert to expected local frame
+  auto toStripFrame = [&](const Vector2D& xy) -> Vector2D {
+    auto shifted = xy + offset;
+    double r = VectorHelpers::perp(shifted);
+    double phi = VectorHelpers::phi(shifted);
+    return Vector2D(r, phi);
+  };
+
+  auto random_point = [&]() -> Vector2D { return {xDist(rng), yDist(rng)}; };
+
+  // for covariance based check, set up one;
+  ActsMatrixD<2, 2> cov;
+  cov << 1.0, 0, 0, 0.05;
+
+  BoundaryCheck bcAbs{true};
+  BoundaryCheck bcTol0{true, false, 1.0, 0};
+  BoundaryCheck bcTol1{false, true, 0, 0.2};
+  BoundaryCheck bcTol01{true, true, 1.0, 0.2};
+  BoundaryCheck bcCov{cov, 1};
+
+  // visualization to make sense of things
+  for (size_t i = 0; i < 10000; i++) {
+    const Vector2D loc{xDist(rng), yDist(rng)};
+    auto locPC = toStripFrame(loc);
+    bool isInsideAbs = aBounds.inside(locPC, bcAbs);
+    bool isInsideTol0 = aBounds.inside(locPC, bcTol0);
+    bool isInsideTol1 = aBounds.inside(locPC, bcTol1);
+    bool isInsideTol01 = aBounds.inside(locPC, bcTol01);
+
+    bool isInsideCov = aBounds.inside(locPC, bcCov);
+
+    os << loc.x() << "," << loc.y() << "," << isInsideAbs << "," << isInsideTol0
+       << "," << isInsideTol1 << "," << isInsideTol01 << "," << isInsideCov
+       << std::endl;
+  }
+
+  // // This point is inside the area
```

Comment:
Is this replaced by the testPoints ?

Then I would just remove the block.

Reply:
Ah yeah. Removed it.